provider/cluster_rosa_classic_resource.go

```diff
@@ -858,22 +906,46 @@ func (r *ClusterRosaClassicResource) Create(ctx context.Context,
 	if response.Diagnostics.HasError() {
 		return
 	}
+	summary := "Can't build cluster"
+
+	version, err := r.getVersion(ctx, state)
+	if err != nil {
+		response.Diagnostics.AddError(
+			summary,
+			fmt.Sprintf(
+				"Can't build cluster with name '%s': %v",
+				state.Name.Value, err,
+			),
+		)
+	}
 
-	err := r.validateAccountRoles(ctx, state)
+	err = r.validateAccountRoles(ctx, state, version)
 	if err != nil {
 		response.Diagnostics.AddError(
-			"Can't build cluster",
+			summary,
 			fmt.Sprintf(
 				"Can't build cluster with name '%s', failed while validating account roles: %v",
 				state.Name.Value, err,
 			),
 		)
 		return
 	}
+	err = validateHttpTokens(ctx, r.logger, state, version)
```

Comment:
Why not validate these things in a Validator. See https://github.com/terraform-redhat/terraform-provider-ocm/blob/main/provider/idps/mapping_method_validator.go and https://github.com/terraform-redhat/terraform-provider-ocm/blob/main/provider/identity_provider_resource.go#L67